### PR TITLE
First branch

### PR DESCRIPTION
**Pull request by RoyVardi4**

### CODE REVIEW

File: package-lock.json
Comment (reuthaselkorn) [inline package-lock.json:1]: add to .gitignore

File: src/api/routes/recipes.ts [hunk 1]
```diff
@@ -4,14 +4,48 @@ import { Router, Request, Response, NextFunction } from "express";
...
+      res.json(response.data);
+    } catch (error) {
+      console.error(error);
+    }
+  })
+
+  route.get('/:id/information', async (req:Request, res:Response) => {
+    const options = {
+      method: 'GET',
+      url: `https://api.spoonacular.com/recipes/${req.params.id}/information`,
```
Comment (reuthaselkorn) [inline src/api/routes/recipes.ts]: site url in a variable

File: src/api/routes/recipes.ts [hunk 2]
```diff
@@ -4,14 +4,48 @@ import { Router, Request, Response, NextFunction } from "express";
...
+      headers: {
+        'x-api-key': 'ee8b40f50b414ed5b735bb593bc5c8d5'
```
Comment (reuthaselkorn) [inline src/api/routes/recipes.ts]: passwords should be in varaiables

File: src/api/routes/recipes.ts [hunk 3]
```diff
@@ -4,14 +4,48 @@ import { Router, Request, Response, NextFunction } from "express";
...
-    return res.send('hi there!') 
+  route.get('/complexSearch', async (req:Request, res:Response) => {
+    const options = {
+      method: 'GET',
+      url: 'https://api.spoonacular.com/recipes/complexSearch',
+      params: {
+        query: req.query.queryString,
+      },
+      headers: {
+        'x-api-key': 'ee8b40f50b414ed5b735bb593bc5c8d5'
```
Comment (reuthaselkorn) [inline src/api/routes/recipes.ts]: passwords should be in varaiables

File: src/api/routes/recipes.ts [hunk 4]
```diff
@@ -4,14 +4,48 @@ import { Router, Request, Response, NextFunction } from "express";
...
 export default (app: Router) => {
-  app.use("/auth", route);
+  app.use("/recipes", route);
 
-  route.get('/', (req, res) => {
-    return res.send('hi there!') 
+  route.get('/complexSearch', async (req:Request, res:Response) => {
+    const options = {
+      method: 'GET',
+      url: 'https://api.spoonacular.com/recipes/complexSearch',
```
Comment (reuthaselkorn) [inline src/api/routes/recipes.ts]: site url in a variable

File: src/api/routes/recipes.ts [hunk 5]
```diff
@@ -4,14 +4,48 @@ import { Router, Request, Response, NextFunction } from "express";
...
+      headers: {
+        'x-api-key': 'ee8b40f50b414ed5b735bb593bc5c8d5'
+      }
+    };
+
+    try {
+      const response = await axios.request(options);
+      res.json(response.data);
+    } catch (error) {
+      console.error(error);
```
Comment (reuthaselkorn) [inline src/api/routes/recipes.ts]: return error to client

File: src/api/routes/recipes.ts [hunk 6]
```diff
@@ -4,14 +4,48 @@ import { Router, Request, Response, NextFunction } from "express";
...
```
Comment (reuthaselkorn) [inline src/api/routes/recipes.ts]: return error to client

File: package-lock.json
Comment (RoyVardi4) [reply package-lock.json:1]: Actually package-lock is very important to be committed for Dependency Version Consistency.

Here a article which explains it nicely [click me!](https://medium.com/@InspireTech/should-i-add-package-lock-json-to-git-why-is-it-important-to-commit-package-lock-json-into-git-e8273a8aad00#:~:text=reliable%20software%20deployments.-,Committing%20the%20package%2Dlock.,dependencies%20used%20in%20a%20project.)

File: src/api/routes/recipes.ts [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+  getMyRecipes,
+  getMyRecipesImages
+} from "../handlers/recipes";
+
+const route = Router();
+
+export default (app: Router) => {
+  app.use("/recipes", route);
+
+  route.get("/complexSearch", getRecipesByComplextQuery);
```
Comment (reuthaselkorn) [inline src/api/routes/recipes.ts:38]: use authentication middleware for all

File: .env
Comment (reuthaselkorn) [inline .env:1]: what about server type and port?

File: .env [hunk 8]
```diff
@@ -0,0 +1,5 @@
+DATABASE_URL = mongodb://localhost:27017
+ACCESS_TOKEN_SECRET=access
```
Comment (reuthaselkorn) [inline .env:2]: should be a stronger password- I'll update with the password I have generated

File: src/lib/express.ts [hunk 9]
```diff
@@ -22,7 +22,10 @@ export default ({ app }: { app: express.Application }) => {
   app.enable("trust proxy");
   app.use(cors());
   app.use(methodOverride());
-  app.use(bodyParser.json());
+  // app.use(bodyParser.json());
```
Comment (reuthaselkorn) [inline src/lib/express.ts:25]: why?

File: src/main.ts [hunk 10]
```diff
@@ -7,11 +7,13 @@ import mongoose from "mongoose";
 async function startServer() {
   const app: express.Application = express();
   await libs.default({ serverApp: app });
-  mongoose.connect(process.env.DATABASE_URL)
-  .then(() => 
-  app &&
-  app.listen(config.port, () => {
-    Logger.info(`
+  mongoose
+    .connect(process.env.DATABASE_URL, { dbName: 'roy-test'})
```
Comment (reuthaselkorn) [inline src/main.ts]: add db name to .env and take it from there

**Pull request merged**